src/PhpWord/Element/Image.php

```diff
@@ -367,7 +387,11 @@ public function getImageStringData($base64 = false)
                 imagesavealpha($imageResource, true);
             }
             ob_start();
-            call_user_func($this->imageFunc, $imageResource);
+            if (!empty($this->imageQuality)) {
```

Comment:
```sh
php -r 'var_dump(empty(0));'

bool(true)
```

So this condition will not apply an imageQuality of 0, instead it will use the second call without specifying a quality.

Also why test for the present of imageQuality at all ? why not pass it in all cases ?

Reply:
Only `imagejpeg()` and `imagepng()` accept a parameter for quality (and for both that happens to be the third parameter). `imagegif()` doesn't have that so you would else pass more parameters to that function than that it defines.
While PHP does allow additional parameters to be passed to a function, I think that we should avoid doing that if it is not necessary.

Reply:
Also, I'm not sure if the default value of a PHP function is kept if you explicitly pass `NULL` to it.

Reply:
The default value for parameter `$quality` in `imagejpeg()` is `-1`: https://www.php.net/imagejpeg

Reply:
damned, I was not aware of how `$this->imageFunc` worked :roll_eyes: 

What I would do is inside `setFunctions()`, instead of affecting a string, create and affect an anonymous function. So we would have anonymous functions for each type of image but they can all be called in this same way.

More or lesse something like:

```php
$this->imageFunc = function ($resource) { imagepng($resource, null, $this->imageQuality); };
// OR 
$this->imageFunc = function ($resource) { imagejpeg($resource, null, $this->imageQuality); };
```

Idem for imageCreateFunc. I realize this is out of scope of this PR, but if you are willing to do that, great ! Otherwise let me know.

Reply:
@PowerKiKi
I would love to refactor it in the way that you are suggesting, but I see that I'm not able to get to that right away.

But I see that the current implementation has its limits and accidentally works just because parameter `$quality` is at both positions for `imagejpeg()` and `imagepng()`.